core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -73,7 +74,7 @@ class BrokerMetadataListener(
   /**
    * The current metadata delta. Accessed only from the event queue thread.
    */
-  private var _delta = new MetadataDelta(_image)
+  private var _delta = new MetadataDelta(_image, metadataVersionProvider)
```

Comment:
The `MetadataImage` should have a metadata version associated with it. So there is no need for the delta to have a "metadata version provider" since the delta already knows what metadata version the image it is modifying is based on.

if the version changes the interpretation of records we would certainly want to make sure that we were using the correct one for the image we're modifying, not something sourced from a global manager object somewhere.

Reply:
The idea with this provider thing was to have an object accessible outside of the metadata update path that could be used to get the current metadata version. Replica fetchers was one consideration. These will need to get the current metadata.version when deciding which RPC to use (that is, until we switch them over to using ApiVersions). However, I suppose ReplicaManager can push the current metadata.version up to ReplicaFetcherManager as updates are processed. 

Come to think of it, It's probably safer that we _don't_ have some out-of-band way to access the metadata.version and instead force components to use MetadataImage (or subscribe to metadata updates themselves). I'll back out this code.